enhancements/network/core-network-policies.md

```diff
@@ -0,0 +1,532 @@
+---
+title: core-nework-policies
+authors:
+  - "@bbennett"
+reviewers:
+  - TBD
+approvers: 
+  - TBD
+api-approvers:
+  - No new API
+tracking-link:
+  - "https://issues.redhat.com/browse/OCPSTRAT-819"
+creation-date: 2024-08-13
+last-updated: 2024-11-14
+---
+
+# Network Policies for OpenShift Components
+
+
+## Summary
+
+Kubernetes NetworkPolicy objects allow control over what traffic can
+enter and leave a pod.  Currently, most OpenShift operators do not
+install policies for their namespaces.  This is a vital tool for
+network security that we should use.
+
+## Motivation
+
+OpenShift should use the OpenShift platform capabilities to secure the
+OpenShift platform workloads (including layered products that are
+considered part of the platform).  It should both restrict what can
+talk to our pods, but also what our pods can talk to, so that a
+compromised pod can not be used as a springboard for other attacks.
+
+
+### User Stories
+
+- As an administrator, I want to have confidence that the cluster
+  components are as secure as possible, and are not likely to be an
+  attack vector so that I can trust the platform.
+
+- As an administrator, I need a way to be able to disable the
+  OpenShift policies in case something goes wrong so that I can
+  operate my cluster as it was before while the policies are being
+  fixed
+
+- As an administrator, I need to be able to override specific
+  OpenShift policies to be more restrictive so that I can satisfy my
+  security department
```

Comment:
I want administrators to create AdminNetworkPolicy to express their exceptions (where they can deny).

Ultimately, I would love for operators to have a way to have config that allows them to create policy to restrict things like webhooks.  They may need an allow-list of valid domains, or something.

But, for the shorter term, we already have people deploying their own Network Policy (not admin, it wasn't ready then) to restrict our stuff since we have no policy today.